docs/clean/developer_guide/stages/execution_order.rst

```diff
@@ -0,0 +1,138 @@
+Execution Order
+===============
+
+Now that we have all the logic defined, we need to specify the order of which the stage steps execute in. The `exec_order` list is the variable used to specify this and will be read first-to-last. The list should contain the method names we wrote in the previous sections. Any methods not in the list will not execute. If the list contains an invalid name (a method that does not exist) a test will fail letting you know.
+
+In this example the methods will execute in the following order:
+
+1. delete_boot_variable
+2. configure_boot_variable
+3. set_configuration_register
+4. write_memory
+5. verify_boot_variable
+6. verify_configuration_register
+
+.. note::
+
+    This documentation did not show the development of the remaining steps, though, they have been added for completeness.
```

Comment:
might be better tell like "not mandatory, but writing method in order as same as `exec_order` as best practice."